src/jukebox/components/player/backends/spotify/interfacing_spotify.py

```diff
@@ -0,0 +1,120 @@
+# Copyright: 2022
+# SPDX License Identifier: MIT License
+import logging
+import os.path
+import os
+import threading
+
+from ruamel import yaml
+
+import jukebox.plugs as plugin
+import jukebox.cfghandler
+
+from spotipy.oauth2 import SpotifyOAuth
+import spotipy
+
+from components.player.backends.spotify.oauth import create_oauth_website
+
+logger = logging.getLogger('jb.spotify')
+cfg = jukebox.cfghandler.get_handler('player')
+
+
+def sanitize(path: str):
+    return os.path.normpath(path).lstrip('./')
+
+
+class SPOTBackend:
+    def __init__(self, player_status, event_loop):
+        self.loop = event_loop
+        self.player_status = player_status
+        self.cache_file = cfg.setndefault('playerspot', 'oauthcache', value='../../shared/spotify/.spotipyoauthcache')
+        self.client_id = cfg.setndefault('playerspot', 'client_id', value='Phoniebox')
+        self.client_secret = cfg.setndefault('playerspot', 'client_secret', value='Phoniebox_secret')
+        self.redirect_uri = cfg.setndefault('playerspot', 'callback_url',
+                                            value='http://localhost:3001')
+
+        spot_scope = "user-read-playback-state,user-modify-playback-state,streaming"
+        self.auth_manager = SpotifyOAuth(open_browser=False, scope=spot_scope, client_id=self.client_id, client_secret=self.client_secret, redirect_uri=self.redirect_uri, cache_path=os.path.abspath(self.cache_file))
+        self.access_token = ""
+        self.spot_client = None
+
+    @plugin.tag
+    def init_spotclient(self, spot_code=None):
+        token_info = self.auth_manager.get_cached_token()
+        token_info = self.auth_manager.validate_token(token_info)
+        logger.debug(f"Token Info: {token_info}")
+
+        if token_info:
+            logger.debug("Found cached token for Spotify Client!")
+            self.access_token = token_info['access_token']
+        else:
+            spotify_oauth_website = create_oauth_website(self.auth_manager)
+            self.thread = threading.Thread(target=spotify_oauth_website.run, kwargs={'host': '', 'port': 3001, 'debug': True})
```

Comment:
Should the host also have a default value, if the port has one?

Reply:
I am not sure if localhost works. Will check